File: frontend/lib/features/files/domain/file_node.dart
/// 空间类型枚举
enum SpaceType {
  personal('PERSONAL'),
  shared('SHARED');

  final String value;
  const SpaceType(this.value);

  static SpaceType fromValue(String value) {
    return SpaceType.values.firstWhere(
      (e) => e.value == value,
      orElse: () => SpaceType.personal,
    );
  }
}

class FileNode {
  const FileNode({
    required this.id,
    required this.parentId,
    required this.name,
    required this.isFolder,
    required this.nodeType,
    required this.normalizedPath,
    required this.sizeBytes,
    required this.updatedAt,
    this.mimeType,
    this.spaceType = SpaceType.personal,
    this.uploadedBy,
    this.mediaAutoImportTaskId,
  });

  final String id;
  final String? parentId;
  final String name;
  final bool isFolder;
  final String nodeType;
  final String normalizedPath;
  final String? mimeType;
  final int sizeBytes;
  final DateTime? updatedAt;
  final SpaceType spaceType;
  final String? uploadedBy;
  final String? mediaAutoImportTaskId;
}

class FileNodePage {
  const FileNodePage({
    required this.items,
    required this.page,
    required this.size,
    required this.totalElements,
    required this.totalPages,
  });

  final List<FileNode> items;
  final int page;
  final int size;
  final int totalElements;
  final int totalPages;

  bool get hasNextPage => page + 1 < totalPages;
}

File: backend/omninest-file/src/main/java/com/omninest/modules/file/domain/FileNode.java
package com.omninest.modules.file.domain;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Id;
import jakarta.persistence.PrePersist;
import jakarta.persistence.PreUpdate;
import jakarta.persistence.Table;
import jakarta.persistence.Version;
import java.time.Instant;
import java.util.UUID;
import com.omninest.modules.file.domain.SourceType;
import com.omninest.modules.file.domain.SpaceType;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "file_nodes", schema = "omni")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class FileNode {
    @Id
    private UUID id;

    @Column(name = "owner_user_id", nullable = false)
    private UUID ownerUserId;

    @Column(name = "parent_id")
    private UUID parentId;

    @Column(name = "node_type", nullable = false, length = 16)
    private String nodeType;

    @Column(nullable = false, length = 255)
    private String name;

    @Column(name = "normalized_path", nullable = false)
    private String normalizedPath;

    @Column(name = "mime_type", length = 160)
    private String mimeType;

    @Column(name = "size_bytes", nullable = false)
    private long sizeBytes;

    @Column(name = "current_object_id")
    private UUID currentObjectId;

    @Column(name = "source_type", nullable = false, length = 32)
    private String sourceType = SourceType.LOCAL.getValue();

    @Column(name = "is_deleted", nullable = false)
    private boolean deleted;

    @Column(name = "deleted_at")
    private Instant deletedAt;

    @Column(name = "deleted_by")
    private UUID deletedBy;

    @Enumerated(EnumType.STRING)
    @Column(name = "purge_state", nullable = false, length = 24)
    private FilePurgeState purgeState = FilePurgeState.NONE;

    @Column(name = "purge_task_id")
    private UUID purgeTaskId;

    @Column(name = "purge_requested_at")
    private Instant purgeRequestedAt;

    @Column(nullable = false)
    private boolean shared;

    @Column(name = "shared_at")
    private Instant sharedAt;

    @Column(name = "created_at", nullable = false)
    private Instant createdAt;

    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

    @Version
    @Column(nullable = false)
    private long version;

    @Enumerated(EnumType.STRING)
    @Column(name = "space_type", nullable = false, length = 20)
    private SpaceType spaceType = SpaceType.PERSONAL;

    @Column(name = "uploaded_by")
    private UUID uploadedBy;

    @PrePersist
    void fillCreatedFields() {
        if (id == null) {
            id = UUID.randomUUID();
        }
        Instant now = Instant.now();
        if (createdAt == null) {
            createdAt = now;
        }
        if (updatedAt == null) {
            updatedAt = now;
        }
    }

    @PreUpdate
    void fillUpdatedAt() {
        updatedAt = Instant.now();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof FileNode other)) return false;
        return id != null && id.equals(other.getId());
    }

    @Override
    public int hashCode() {
        return getClass().hashCode();
    }
}
